Spell_Chech_Example.c

```diff
@@ -38,43 +39,67 @@ _spell_check_word(const char* str)
 
 
 /************ EFL Spell Check Logic ******************/
+
+static Efl2_Text_Attribute_Handle*
+mark_misspelled(Eo *start,Eo *end)
+{
+   efl2_text_style_underline_color_set(attribute_factory, 255, 0, 0, 255);
+   Efl2_Text_Attribute_Handle handle = efl2_text_attribute_factory_insert(attribute_factory, start, end);
+   Efl2_Text_Attribute_Handle *handle_ref = malloc(sizeof(Efl2_Text_Attribute_Handle));
+   memcpy(ref_handle,&handle,sizeof(Efl2_Text_Attribute_Handle));
+   efl2_text_attribute_factory_ref(handle_ref);
+   return handle_ref;
+}
+
+static void
+mark_clear()
+{
+   Eina_List *l;
+   EINA_LIST_FOREACH(handles_list, l, handle)
+     {
+        efl2_text_attribute_factory_unref(handle);
+        efl2_text_attribute_factory_del(handle);
+        free(handle);
+     }
+   handles_list = eina_list_free(handles_list);
+}
+
 static Eina_Bool
 _ui_text_spell_check_cb(void *data, const Efl_Event *event EINA_UNUSED)
```

Comment:
Worth adding a comment somewhere that this is a *very* naive implementation. Under normal circumstances you'd probably not clear everything, but rather only clear the changed ones and re-add the changed ones.
You'll probably also only add them in the viewport or twice the viewport or whatever, and not annotate the whole file.
Just worth mentioning probably.

Reply:
This comment still stands

Reply:
Sorry, I do not understand what you mean

Reply:
What you are doing here, for simplicity, is every time removing all of the handles and then adding them again. In a real implementation you won't actually check the whole text for spelling mistakes every time, and more than that, even if you did, you wouldn't need to remove all the handles and add them again, you'd only need to change the bad ones.

For example for this text (bold is for misspelling):

"I **wnet** to the hot", when you add "e" it should become "I  **wnet** to the **hote**" because "hote" is not a word and hot is. However, because only one character changed you don't need to examine the whole text again for spelling mistakes, and you don't need to remove/change the one for "wnet". It's fine in this small contained example, but it's worth it to leave a note that this is not efficient for large bodies of text and you don't actually have to remove all of the handles.

Reply:
You are talking about auto spell check.
For simplicity sake of this example, We do spell check on request (not automatic/interactive)

We can do other example for that

Reply:
I know it's not doing auto spellcheck, and that's absolutely fine for this example (and was even in my original description of this task).

All I'm talking about is adding a *comment* in the code about how this is the easiest way of doing this for small chunks of texts, but if you want to implement spell checking for large chunks of texts you'll need to implement more advanced strategies.

Reply:
Add comments about it.
We will try to create advance example on how to deal with attributes handles (like auto spelling Check)